analyses/focal-cn-file-preparation/01-prepare-cn-file.R

```diff
@@ -0,0 +1,113 @@
+# This script converts a seg file into a tsv file with CN information and gene
+# annotation.
+#
+# Code adapted from the PPTC PDX Focal Copy NUmber and SV Plots repository here:
+# https://github.com/marislab/pptc-pdx-copy-number-and-SVs/blob/master/R/focal-CN-revision.R
+#
+# Chante Bethell for CCDL 2019 and Jo Lynne Rokita
+#
+# #### USAGE
+# This script is intended to be run via the command line from the top directory
+# of the repository as follows:
+#
+# Rscript 'analyses/oncoprint-landscape/01-prepare-cn-file.R'
+
+#### Set Up --------------------------------------------------------------------
+
+# Install GenomicRanges
+if (!("GenomicRanges" %in% installed.packages())) {
+  install.packages("BiocManager")
+  BiocManager::install("GenomicRanges")
+}
+
+# Install IRanges
+if (!("IRanges" %in% installed.packages())) {
+  install.packages("BiocManager")
+  BiocManager::install("Iranges")
+}
+
+# Install annotatr 
+if (!("annotatr" %in% installed.packages())) {
+  install.packages("BiocManager")
+  BiocManager::install("annotatr")
+}
+
+# Get `magrittr` pipe
+`%>%` <- dplyr::`%>%`
+
+#### Command line options ------------------------------------------------------
+
+# Declare command line options
+option_list <- list(
+  optparse::make_option(
+    c("-f", "--seg_file"),
+    type = "character",
+    default = NULL,
+    help = "file path to SEG file that contains cnv information"
+  )
+)
+
+# Read the arguments passed
+opt_parser <- optparse::OptionParser(option_list = option_list)
+opt <- optparse::parse_args(opt_parser)
+
+#### Directories and Files -----------------------------------------------------
+
+results_dir <- "results"
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+
+seg_file <-
+  data.table::fread(
+    file.path(opt$seg_file),
+    data.table = FALSE,
+    stringsAsFactors = FALSE
+  )
+
+#### Format seg file and overlap with hg38 genome annotations ------------------
+
+# Exclude the X and Y chromosomes
+seg_no_xy <- subset(seg_file, chrom != "Y" & chrom != "X")
+seg_no_xy <- seg_no_xy[, c(2:ncol(seg_no_xy), 1)]
+# names(seg_no_xy) <- c("chr", "start", "end", "markers", "CN", "Model")
+
+# Make seg data.frame a GRanges object 
+seg_gr <- seg_no_xy %>%
+  GenomicRanges::makeGRangesFromDataFrame(keep.extra.columns = TRUE,
+                                          starts.in.df.are.0based = FALSE)
+
+# Define the annotations for the hg38 genome
+annotations <- annotatr::build_annotations(genome = "hg38",
+                                           annotations = "hg38_basicgenes")
```

Comment:
```suggestion
                                           annotations = "hg38_genes_exons")
```

This should speed up this and the next step